File: navigation/src/main/java/com/example/navigation/stepsEngine/field/AcquirerField.java
package com.example.navigation.stepsEngine.field;

import com.example.navigation.stepsEngine.enums.Acquirer;
import com.example.navigation.stepsEngine.payment.FlowState;

public class AcquirerField extends Field {

    @Override
    public String get(FlowState flowState) {
        final Acquirer acquirer = flowState.getAcquirer();
        if (acquirer == null) {
            return null;
        } else {
            return acquirer.name();
        }
    }

    @Override
    public String getId() {
        return null;
    }

    @Override
    public void set(FlowState flowState, Object value) {

    }

}

File: navigation/src/main/java/com/example/navigation/stepsEngine/field/AcquirerField.kt
package com.example.navigation.stepsEngine.field

import com.example.navigation.stepsEngine.payment.FlowState
import kotlinx.android.parcel.Parcelize

@Parcelize
class AcquirerField : Field() {
    override operator fun get(flowState: FlowState): String? {
        val acquirer = flowState.acquirer
        return acquirer?.name
    }

    override fun getId(): String {
        return FieldName.ACQUIRER
    }

    override operator fun set(flowState: FlowState, value: Any?) {}
}